src/util/resolve.ts

```diff
@@ -0,0 +1,21 @@
+import { hash, API } from "../api";
+
+const _resolutionTable: Map<string, hash> = new Map()
+
+// returns an instanceHash based on an sid
+// throw an error if the sid doesn't exists or have if it has more or less than one instance running
+export const resolveSID = async (api: API, sid: string): Promise<hash> => {
+  if (_resolutionTable.has(sid)) return _resolutionTable.get(sid)
+
+  const { services } = await api.service.list({})
+  const service = services.find(x => x.sid === sid)
```

Comment:
same issue as before, they can be multiple service with the same sid. a filter should be used:
```suggestion
  const services = services.filter(x => x.sid === sid)
  if (services.length > 1) throw new Error(`multiple services resolve ${sid}`)
  const service = services.pop()
```

Reply:
done thx